### PR TITLE
Update tests to use react testing library

### PR DESCRIPTION
**Pull request by culshaw**
This also removes a series of dead dependencies. 
Updates the codebase to use react v18.

### CODE REVIEW

**Comment by culshaw** [comment]
Also, we're happy to take over the publishing of this repo if you haven't the time to maintain it @ianmcgregor ?

**Review by ianmcgregor — APPROVED** [review]
Great stuff, thanks for keeping this alive and improving it so much 🙌

File: package.json [hunk 1]
```diff
@@ -11,35 +11,37 @@
...
+    "testEnvironment": "jsdom",
+    "testEnvironmentOptions": {
+      "url": "http://localhost"
+    }
   },
   "license": "ISC",
   "repository": {
     "type": "git",
-    "url": "https://github.com/ianmcgregor/react-micro-router"
+    "url": "https://github.com/yunojuno/react-micro-router"
```
Comment (ianmcgregor) [inline package.json]: this got copied over from your fork

File: shell.nix [hunk 2]
```diff
@@ -0,0 +1,89 @@
+# YunoJuno's Dev Environment
```
Comment (ianmcgregor) [inline shell.nix]: looks like the nix stuff got added to the commit by mistake?

**Review by ianmcgregor — DISMISSED** [review]
I think just a couple bits added by mistake here

**Comment by culshaw** [comment]
@ianmcgregor This has been fixed 👍

**Review by ianmcgregor — APPROVED** [review]

**Pull request merged**